unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/VRMExport/VRMDetailsController.cs

```diff
@@ -0,0 +1,133 @@
+using Cysharp.Threading.Tasks;
+using DCL.Helpers;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace DCL.Backpack
+{
+    public class VRMDetailsController : IDisposable
+    {
+        private readonly IUserProfileBridge userProfileBridge;
+        private readonly IVRMDetailsComponentView view;
+        private readonly Dictionary<string, NFTDataDTO> nftDataCache = new ();
+
+        public event Action<string, UnequipWearableSource> OnWearableUnequipped;
+        public event Action<string, EquipWearableSource> OnWearableEquipped;
+
+        public VRMDetailsController(
+            IVRMDetailsComponentView view,
+            IUserProfileBridge userProfileBridge)
+        {
+            this.view = view;
+            this.userProfileBridge = userProfileBridge;
+
+            view.OnWearableUnequipped += HandleWearableUnequipped;
+            view.OnWearableEquipped += HandleWearableEquipped;
+        }
+
+        public void Dispose()
+        {
+            view.OnWearableUnequipped -= HandleWearableUnequipped;
+            view.OnWearableEquipped += HandleWearableEquipped;
+        }
+
+        private void HandleWearableEquipped(VRMItemModel vrmItemModel, EquipWearableSource equipWearableSource)
+        {
+            OnWearableEquipped?.Invoke(vrmItemModel.wearableUrn, equipWearableSource);
+        }
+
+        private void HandleWearableUnequipped(VRMItemModel vrmItemModel, UnequipWearableSource unEquipWearableSource)
+        {
+            OnWearableUnequipped?.Invoke(vrmItemModel.wearableUrn, unEquipWearableSource);
+        }
+
+        public void Initialize(List<string> vrmBlockingWearablesList)
+        {
+            PopulateDetails(vrmBlockingWearablesList).Forget();
+        }
+
+        private async UniTask PopulateDetails(List<string> vrmBlockingWearablesList)
+        {
+            List<NFTDataDTO> itemsToDisplay = new ();
+            // Copy the list from params since it's used for other checks elsewhere
+            List<string> itemsToProcess = new (vrmBlockingWearablesList);
+
+            // Get cached items first, to reduce api calls.
+            for (int i = itemsToProcess.Count - 1; i >= 0; i--)
+                if (nftDataCache.TryGetValue(itemsToProcess[i], out NFTDataDTO itemData))
+                {
+                    itemsToDisplay.Add(itemData);
+                    itemsToProcess.RemoveAt(i);
+                }
+
+            // If we're still missing items, fetch them.
+            if (itemsToProcess.Count > 0)
+            {
+                string nftItemRawData = await WearablesFetchingHelper.GetNFTItems(itemsToProcess);
```

Comment:
I think that making a static call here is not desirable. It makes impossible to properly make unit tests for the class

Reply:
I get your point, but `WearablesFetchingHelper` isn't interfaced nor used in any other class that implements an interface I can use for injection in this controller and later mock in tests. I take you suggestion and note it as a tech debt for a later PR 👌